traitlets/traitlets.py

```diff
@@ -1041,11 +1052,16 @@ def unobserve(self, handler, names=All, type='change'):
             The names of the traits for which the specified handler should be
             uninstalled. If names is All, the specified handler is uninstalled
             from the list of notifiers corresponding to all changes.
+        tags: dict
+            Removes the handler from all traits whose metadata matches the given
+            tags or selector functions (`HasTraits.traits` for selector info)
         type : str or All (default: 'change')
             The type of notification to filter by. If All, the specified handler
             is uninstalled from the list of notifiers corresponding to all types.
         """
         names = parse_notifier_name(names)
+        if tags:
+            names.extend(self.trait_names(**tags))
```

Comment:
If I understand this well, if `names` is  `All`, then you end up listening to the All event *and*  the filtered tags.